bat-utils/lib/runtime-wallet.js

```diff
@@ -123,6 +123,34 @@ Wallet.prototype.providers = function () {
   return underscore.keys(Wallet.providers)
 }
 
+Wallet.prototype.isGrantExpired = function (info, grant) {
+  const { token } = grant
+
+  const jws = braveUtils.extractJws(token)
+  const { expiryTime } = jws
+
+  return Date.now() > (expiryTime * 1000)
+}
+
+Wallet.prototype.expireGrant = async function (info, wallet, grant) {
+  const { runtime } = this
+  const { database } = runtime
+  const { paymentId } = wallet
+  const { token } = grant
+
+  const wallets = database.get('wallets', debug)
+
+  const $set = {
+    'grants.$.status': 'expired'
+  }
+  const state = { $set }
+  const where = {
+    paymentId,
+    'grant.$.token': token
```

Comment:
I would suggest changing this to filter on the `grantId` to be consistent with:
https://github.com/brave-intl/bat-ledger/blob/master/ledger/controllers/wallet.js#L300